source/2_platform/rex_windows/src/gui_application.cpp

```diff
@@ -5,41 +5,186 @@
+      {
+        release_device_context(hwnd, s_glctx.dc);
+        REX_ERROR("Failed to create the initial rendering context");
+        return false;
+      }
+
+      wglMakeCurrent(s_glctx.dc, temp);
+
+      // Load WGL Extensions:
+      gladLoaderLoadWGL(s_glctx.dc);
+
+      s32 attribs[] =
+      {
+        WGL_CONTEXT_MAJOR_VERSION_ARB, 3,       // Set the MAJOR version of OpenGL to 3
+        WGL_CONTEXT_MINOR_VERSION_ARB, 2,       // Set the MINOR version of OpenGL to 2
+        WGL_CONTEXT_FLAGS_ARB, 
+        WGL_CONTEXT_FORWARD_COMPATIBLE_BIT_ARB, // Set our OpenGL context to be forward compatible
+        0
+      };
+
+      const char* extensions = (const char*)wglGetExtensionsStringARB(s_glctx.dc);
+      if (strstr(extensions, "WGL_ARB_create_context") == NULL)
+      {
+        s_glctx.glrc = wglCreateContextAttribsARB(s_glctx.dc, NULL, attribs);
+        if (s_glctx.glrc == NULL)
+        {
+          wglDeleteContext(temp);
+          release_device_context(hwnd, s_glctx.dc);
+          REX_ERROR("Failed to create Windows OpenGL Context");
+
+          return false;
+        }
+
+        wglMakeCurrent(NULL, NULL);
+        wglDeleteContext(temp);
+        wglMakeCurrent(s_glctx.dc, s_glctx.glrc);
+      }
+      else
+      {
+        s_glctx.glrc = temp;
+      }
+
+      if (!gladLoaderLoadGL()) 
+      {
+        wglMakeCurrent(NULL, NULL);
+        wglDeleteContext(s_glctx.glrc);
+        release_device_context(hwnd, s_glctx.dc);
+        REX_ERROR("Glad Loader failed!");
+        
+        return false;
+      }
+
+      return true;
+    }
+  }
+}
+
+#define create_ctx(hwnd) rex::opengl::create_gl_context(hwnd)
+#else
+#define create_ctx(hwnd) (hwnd == nullptr)
+#endif
+
 namespace rex
 {
   namespace win32
   {
     struct GuiApplication::Internal
     {
       Internal(const PlatformCreationParams& guiCreationParams, const ApplicationCreationParams& appParams, CommandLineArguments&& genericCreationParams)
-          : platform_creation_params(rsl::move(guiCreationParams))
-          , app_params(appParams)
-          , cmd_line_args(rsl::move(genericCreationParams))
+        : platform_creation_params(rsl::move(guiCreationParams))
+        , app_params(appParams)
+        , cmd_line_args(rsl::move(genericCreationParams))
       {
       }
 
       bool initialize()
       {
         window = create_window();
-        if(window == nullptr)
+        if (window == nullptr)
+        {
+          return false;
+        }
+
+        HWND hwnd = (HWND)window->get_primary_display_handle();
+        if (create_ctx(hwnd) == false)
+        {
+          return false;
+        }
+
+        if (renderer::initialize((void*)&hwnd, app_params.max_render_commands) == false)
         {
           return false;
         }
 
+        RendererInfo info = renderer::get_info();
+        REX_INFO("Renderer Info - API Version: {}", info.api_version);
+        REX_INFO("Renderer Info - Renderer: {}", info.renderer);
```

Comment:
This returns the GPU name, not the "renderer" name (renderer is called cera).
so maybe change the name to "gpu_model"?

Reply:
I understand your comment here and to me, it also is a bit confusing but in the end, the GPU is the one that renders everything on screen ... not our class it just manages the instructions that need to be scheduled and executed. You even call glRenderer to retrieve this information.

I would say that it might be a better idea to call it adaptor or something as within DX this is what those things are called IIRC